plugins/com.google.cloud.tools.eclipse.appengine.libraries.ui/src/com/google/cloud/tools/eclipse/appengine/libraries/ui/CloudLibrariesPage.java

```diff
@@ -101,23 +106,33 @@ public void initialize(IJavaProject project, IClasspathEntry[] currentEntries) {
     if (libraries == null || libraries.isEmpty()) {
       return null;
     }
-  
+
     try {
       if (MavenUtils.hasMavenNature(project.getProject())) {
         BuildPath.addMavenLibraries(project.getProject(), libraries, new NullProgressMonitor());
         return new IClasspathEntry[0];
       } else {
+        Set<LibraryFile> masterFiles = new HashSet<>();
         for (Library library : libraries) {
           if (!library.isResolved()) {
             library.resolveDependencies();
           }
+          masterFiles.addAll(library.getLibraryFiles());
         }
+
+        Library masterLibrary = CloudLibraries.getMasterLibrary();
+        masterFiles.addAll(masterLibrary.getLibraryFiles());
+        masterLibrary.setLibraryFiles(masterFiles);
+        ArrayList<Library> masterLibraries = new ArrayList<>();
+        masterLibraries.add(masterLibrary);
```

Comment:
- `List<Library> ... = Collections.singletonList(masterContainer);` reads it better to me.
- The var name `masterLibraries` makes me a bit confused, as it sounds like there are multiple masters.

Reply:
Passing ImmutableLists to methods that don't specifically take an ImmutableList in their signature makes me nervous.

I've refactored so masterLibraries no longer appears here.